async-profiler-context/src/main/java/io/pyroscope/labels/Ref.java

```diff
@@ -16,7 +16,8 @@ public Ref(T val, Long id) {
     public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
-        Ref<T> valueRef = (Ref<T>) o;
+
+        Ref<?> valueRef = (Ref<?>) o;
```

Comment:
Note this uses `?` instead of `T`. I gather that `?` is somewhat safer when casting generic classes.